From c50f313121ddd5179123574b8b27c641b32388e6 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Fri, 30 Nov 2012 22:00:21 +0100
Subject: [PATCH 002/628] zynq: pcie: AXI PCIE Fix section mismatch

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 2786347415bd185eced00fdc8ed8a01bba6498de

Fix section mismatch because of section warnings.

Warning:
WARNING: arch/arm/mach-zynq/built-in.o(.text+0x2c5c):
 Section mismatch in reference from the function xaxi_pcie_setup()
 to the function .devinit.text:xaxi_pcie_set_bridge_resource()
The function xaxi_pcie_setup() references
the function __devinit xaxi_pcie_set_bridge_resource().
This is often because xaxi_pcie_setup lacks a __devinit
annotation or the annotation of xaxi_pcie_set_bridge_resource is wrong.

WARNING: arch/arm/mach-zynq/built-in.o(.text+0x2c68):
 Section mismatch in reference from the function xaxi_pcie_setup()
 to the function .devinit.text:xaxi_pcie_process_bridge_OF_ranges()
The function xaxi_pcie_setup() references
the function __devinit xaxi_pcie_process_bridge_OF_ranges().
This is often because xaxi_pcie_setup lacks a __devinit
annotation or the annotation of xaxi_pcie_process_bridge_OF_ranges is wrong.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 arch/arm/mach-zynq/xaxipcie.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-zynq/xaxipcie.c b/arch/arm/mach-zynq/xaxipcie.c
index b8e07f2..c4b216e 100644
--- a/arch/arm/mach-zynq/xaxipcie.c
+++ b/arch/arm/mach-zynq/xaxipcie.c
@@ -610,7 +610,7 @@ static struct pci_ops xaxi_pcie_ops = {
 	.write = xaxi_pcie_write_config,
 };
 
-static int xaxi_pcie_setup(int nr, struct pci_sys_data *sys)
+static int __devinit xaxi_pcie_setup(int nr, struct pci_sys_data *sys)
 {
 	u32 val;
 	struct xaxi_pcie_port *port = &xaxi_pcie_ports[nr];
-- 
1.7.5.4

